src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -548,24 +548,33 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
 
             var process = CreateRunnableExternalProcess(config);
 
+            var stdout = "";
+            var stderr = "";
+
+            process.OutputDataReceived += (sender, args) => stdout += args.Data;
+            process.ErrorDataReceived += (sender, args) => stderr += args.Data;
+
             try
             {
                 process.Start();
+                process.BeginOutputReadLine();
+                process.BeginErrorReadLine();
             }
             catch (Exception ex)
             {
                 throw new KubeConfigException($"external exec failed due to: {ex.Message}");
             }
 
-            var stdout = process.StandardOutput.ReadToEnd();
-            var stderr = process.StandardError.ReadToEnd();
-            if (string.IsNullOrWhiteSpace(stderr) == false)
+            // Wait for a maximum of 5 seconds, if a response takes longer probably something went wrong...
+            if (!process.WaitForExit(5000) && string.IsNullOrWhiteSpace(stderr))
```

Comment:
We should perhaps wait even longer if the terminal is interactive. Something like this:

https://stackoverflow.com/questions/45393001/determine-if-net-core-console-application-is-running-in-user-interactive-mode

It feels like 5s is pretty fast to get to a web browser and get logged in.

Reply:
When Kubelogin asks for an interactive prompt, it will return an error. However, the process will not be killed.